### PR TITLE
feat(api): set up google cloud profiler

### PR DESCRIPTION
**Pull request by cuixq**
https://github.com/google/osv.dev/issues/2793

This PR attempts to set up Google Cloud Profiler following this [guide](https://cloud.google.com/profiler/docs/profiling-python).

### CODE REVIEW

**Comment by another-rex** [comment]
Related https://github.com/google/osv.dev/pull/1305 . I've added google cloud profiler in that PR before, it didn't work that well when used with appengine, though maybe now that it is running in cloud run it will work better.

Looking at the commit history, it looks like you should use the wheel version: https://github.com/google/osv.dev/pull/1305/commits/5ef22aaab708df7fc91f0472c603caf3c7ea1018

**Comment by cuixq** [comment]
since we are with Cloud Run instead of App Engine now, maybe let's try with the default version first?

**Review by michaelkedar — APPROVED** [review]

**Review by another-rex — APPROVED** [review]

File: gcp/api/server.py [hunk 1]
```diff
@@ -1395,6 +1397,13 @@ def main():
     setup_gcp_logging('api-backend')
     logging.getLogger().addFilter(trace_filter)
 
+    # Profiler initialization. It starts a daemon thread which continuously
+    # collects and uploads profiles. Best done as early as possible.
+    try:
+      googlecloudprofiler.start(service="osv-api-profiler")
```
Comment (hogo6002) [inline gcp/api/server.py:1403]: Do you want to enable this only on the test instance first and then enable it on prod after verification? `if get_gcp_project() == _TEST_INSTANCE:`

Comment (cuixq) [reply gcp/api/server.py:1403]: ah! thanks!

**Pull request merged**